lib-mblast-utils/src/main/kotlin/mblast/util/io/LoggingOutputStream.kt

```diff
@@ -0,0 +1,66 @@
+package mblast.util.io
+
+import io.foxcapades.lib.kps.kpd.UByteDeque
+import java.io.OutputStream
+
+private const val ASCII_NUL: UByte = 0u
+private const val ASCII_LF: UByte = 10u
+private const val ASCII_CR: UByte = 13u
+
+/**
+ * OutputStream implementation that outputs lines of text to a logger function.
+ *
+ * **Example**
+ * ```
+ * // Some text to log
+ * val myText = """
+ * Hello
+ * Goodbye
+ * """
+ *
+ * // Create a log stream wrapping a logger method
+ * val logStream = LoggingOutputStream(logger::debug)
+ *
+ * // Pipe input stream to the logging output stream
+ * myText.byteInputStream().transferTo(logStream)
+ * ```
+ *
+ * @author Elizabeth Paige Harper - https://github.com/foxcapades
+ *
+ * @constructor Constructs a new `LoggingOutputStream` instance.
+ *
+ * @param logWriter Function that will be called with collected log lines.
+ */
+@Suppress("NOTHING_TO_INLINE")
+class LoggingOutputStream(private val logWriter: (line: String) -> Unit) : OutputStream() {
+  private val buffer = UByteDeque(256)
```

Comment:
Maybe comment that this class is not threadsafe (like a normal log library) and that individual threads should create their own instances to do work just for that thread.  I feel like a developer might see a class like this and think they can write: `private static LoggingOutputStream LOGOUT_INFO = new LoggingOutputStream(s -> Logger.getLogger(ThisClass.class).info(s));` or similar as a replacement for a static Logger.  Doing so might cause conflated log messages.

Reply:
Not sure what you're using this for but maybe implementing Writer would be better?  a write(String) would be similar to Logger::info(String).  If you're transferring from an InputStream you could wrap in an InputStreamReader and use a reader -> writer transferTo method.

Reply:
I'm in a wonky situation where I only have access to an InputStream that I need to pass to the platform.

My plan was to tee the InputStream to write out to an OutputStream as it was read.  Then I would use this class as that OutputStream to capture the stderr output as it was read by the platform.